src/worker/plugins/loadPlugin.ts

```diff
@@ -94,3 +102,51 @@ export async function loadPlugin(server: PluginsServer, pluginConfig: PluginConf
     }
     return false
 }
+
+async function inferPluginCapabilities(server: PluginsServer, pluginConfig: PluginConfig, prevConfig?: PluginConfig) {
+    // infer on load implies there's no lazy loading, but all workers get
+    // these properties loaded
+    const vm = await pluginConfig.vm?.resolveInternalVm
+    const capabilities: Capabilities = { scheduled_tasks: [], jobs: [], methods: [] }
```

Comment:
Tip: we should normally avoid using `!` in real (non-test) code. It's a place for subtle bugs, and better to just avoid. Often times, the solution is as simple as replacing `!` with `?`. Here though I'd suggest casting this object to `Required<Capabilities>` instead of just `Capabilities`. Then all the `methods`, `jobs`, etc will be there.